### PR TITLE
store: check constraint for "Delete-Your-Writes" records when txn commit

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve? 

in #9127, we push constraint check down to tikv layer when prewrite phase.

but it miss the situation that txn add record a then delete record a, we also need check record is real duplicate when commit.

so just as `TestDeferConstraintCheckForDelete`, it will lead to "delete wrong data"(primary key) or "inconsist index data"(unique key) 

### What is changed and how it works?

we can do a addition batch get check for those key before commit just like pre-#9127 logic
but maybe better to follow #9127's idea just check it with prewrite mutation flag.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

- impl change
- kvproto
- tikv/unistore

Side effects

 - n/a

Related changes

 - Need to cherry-pick to the release branch 3.0/3.1

Release note

 - fix missing check constraint for "Delete-Your-Writes" records.

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/14968)

### CODE REVIEW

**Comment by lysu** [comment]
need merge https://github.com/pingcap/kvproto/pull/565 first to let CI happy

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -1170,6 +1168,18 @@ func (c *twoPhaseCommitter) execute(ctx context.Context) (err error) {
 	return nil
 }
 
+func (c *twoPhaseCommitter) stripCheckKeys() {
+	var i int
```
Comment (coocood) [inline store/tikv/2pc.go]: byPass if `len(c.checkKeys) == 0`?

**Comment by lysu** [comment]
check_dev_2 fail due to it only run with tikv master by design, it should be green after kv pr merged.

File: store/tikv/2pc.go [hunk 2]
```diff
@@ -264,13 +266,20 @@ func (c *twoPhaseCommitter) initKeysAndMutations() error {
 			}
 			putCnt++
 		} else {
+			var op pb.Op
+			if c := txn.us.GetKeyExistErrInfo(k); c != nil {
+				op = pb.Op_CheckNotExists
+				checkKeys[string(k)] = struct{}{}
```
Comment (youjiali1995) [inline store/tikv/2pc.go]: Suggest adding comments here.

**Comment by coocood** [comment]
LGTM

File: store/tikv/2pc.go [hunk 3]
```diff
@@ -264,13 +266,20 @@ func (c *twoPhaseCommitter) initKeysAndMutations() error {
 			}
 			putCnt++
 		} else {
+			var op pb.Op
+			if c := txn.us.GetKeyExistErrInfo(k); c != nil {
```
Comment (youjiali1995) [inline store/tikv/2pc.go]: Can we eliminate it in the pessimistic transaction?

**Comment by coocood** [comment]
LGTM

**Pull request merged**